Intake testing

eddiert1239 opened the pull request:
feeder motors commented out right now for testing, so this should not be merged (yet)
also includes code in AllInCommand to reject gamepieces after one is intook

eddiert1239 committed:
commented out everything but the intake motors

eddiert1239 committed:
no more following

eddiert1239 committed:
Merge branch 'IntakeTesting' of https://github.com/eddiert1239/Crescendo2024 into IntakeTesting

eddiert1239 committed:
uncommented index

eddiert1239 committed:
changed from intake stopping to reversing

also spotless

eddiert1239 committed:
changed controls to reflect what actually happens

eddiert1239 committed:
Merge branch 'robototes:main' into IntakeTesting

bobtseattle commented on src/main/java/frc/team2412/robot/Hardware.java:
```diff
@@ -27,7 +27,7 @@ public class Hardware {
 	public static final int INDEX_MOTOR_UPPER = 45;
 
-	public static final int FEEDER_MOTOR = 46;
+	// public static final int FEEDER_MOTOR = 46;
```
It would be good to add a comment to know this change needs to be reverted before merge. 
Ditto for all comments.

bobtseattle commented on src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java:
```diff
@@ -15,20 +15,21 @@ public AllInCommand(IntakeSubsystem intakeSubsystem) {
 	public void execute() {
 		if (intakeSubsystem.getIndexSensor()) {
-			intakeSubsystem.intakeStop();
+			intakeSubsystem.intakeReverse();
```
Instead of doing reverse, it would be good to have a reject that runs at a lower speed that doesn't use as much power and reduces the chance of overheating the motors.

bobtseattle commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -22,8 +22,8 @@ public class IntakeSubsystem extends SubsystemBase {
-	public static final double FEEDER_IN_SPEED = 0.3;
-	public static final double FEEDER_REVERSE_SPEED = -0.3;
+	// public static final double FEEDER_IN_SPEED = 0.3;
+	// public static final double FEEDER_REVERSE_SPEED = -0.3;
```
Would be good to have a reject speed constant in addition to the reverse speed.

bobtseattle commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -122,30 +128,35 @@ public void intakeStop() {
+		indexSet(setIndexInSpeedEntry.getDouble(INDEX_IN_SPEED));
 	}
 
 	public void indexReverse() {
```
Seems like we will want a indexReject method.

bobtseattle reviewed (changes requested):
It would be good to get the reject logic created so intake out speed will depend on situation.

eddiert1239 committed:
addressed comments

added comments on what needs to be uncommented before merge
added intakeReject method
changed allin to use reject instead of reverse

eddiert1239 committed:
Merge branch 'robototes:main' into IntakeTesting

eddiert1239 committed:
commented out feeder int fulltargetcommand

eddiert1239 committed:
index upper and lower can have different speeds

eddiert1239 committed:
added @Override

we forgot some and it had some warnings oops

eddiert1239 committed:
added alignment for the feeder and temp logging

eddiert1239 committed:
removed alignment and added feederShoot

eddiert1239 committed:
uncommented feeder

bobtseattle commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -60,6 +67,34 @@ public class IntakeSubsystem extends SubsystemBase {
 
+	private final GenericEntry intakeMotorFrontTemp =
+			Shuffleboard.getTab("Intake")
+					.add("Back Intake temp", 0)
```
Back or front?

bobtseattle reviewed (changes requested):
Back is not front.

eddiert1239 committed:
back is not front.

bobtseattle reviewed (dismissed):
LGTM

eddiert1239 committed:
temp logging for all motors

bobtseattle reviewed (approved):
LGTM
